Rules/AvoidGlobalAliases.cs

```diff
@@ -0,0 +1,195 @@
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    class AvoidGlobalAliases : AstVisitor, IScriptRule
+    {
+        private List<DiagnosticRecord> records;
+        private string fileName;
+
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null)
+            {
+                throw new ArgumentNullException(Strings.NullAstErrorMessage);
+            }
+
+            records = new List<DiagnosticRecord>();
+            this.fileName = fileName;
+
+            ast.Visit(this);
+
+            return records;
+        }
+
+        #region VisitCommand functions
+        public override AstVisitAction VisitCommand(CommandAst commandAst)
+        {
+            if (!IsNewAliasCmdlet(commandAst))
+            {
+                return AstVisitAction.SkipChildren;
+            }
+
+            return AstVisitAction.Continue;
+        }
+
+        public override AstVisitAction VisitCommandParameter(CommandParameterAst commandParameterAst)
+        {
+            if (IsScopeParameterForNewAliasCmdlet(commandParameterAst))
+            {
+                if ((commandParameterAst.Argument != null)   // if the cmdlet looks like -Scope:Global check Parameter.Argument
+                    && (commandParameterAst.Argument.ToString().Equals("Global", StringComparison.OrdinalIgnoreCase)))
+                {
+                    records.Add(new DiagnosticRecord(
+                                    string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalAliasesError),
+                                    commandParameterAst.Extent,
+                                    GetName(),
+                                    DiagnosticSeverity.Warning,
+                                    fileName,
+                                    commandParameterAst.ParameterName));
+                }
+                else
+                {
+                    var nextAst = FindNextAst(commandParameterAst);
+
+                    if ((nextAst is StringConstantExpressionAst)   // if the cmdlet looks like -Scope Global
+                        && (((StringConstantExpressionAst)nextAst).Value.ToString().Equals("Global", StringComparison.OrdinalIgnoreCase)))
+                    {
+                        records.Add(new DiagnosticRecord(
+                                string.Format(CultureInfo.CurrentCulture, Strings.AvoidGlobalAliasesError),
+                                ((StringConstantExpressionAst)nextAst).Extent,
+                                GetName(),
+                                DiagnosticSeverity.Warning,
+                                fileName,
+                                ((StringConstantExpressionAst)nextAst).Value));
+                    }
+                }
+            }
+
+            return AstVisitAction.SkipChildren;
+        }
+        #endregion
+
```

Comment:
I would recommend documentation comment even for private methods that not trivial.